GVFS/GVFS.Common/NamedPipes/NamedPipeServer.cs

```diff
@@ -123,6 +124,13 @@ private void OnNewConnection(IAsyncResult ar, bool createNewThreadIfSynchronous)
                     metadata.Add(TracingConstants.MessageKey.WarningMessage, "OnNewConnection: Connection broken");
                     this.tracer.RelatedEvent(EventLevel.Warning, "OnNewConnectionn_EndWaitForConnection_IOException", metadata);
                 }
+                catch(SocketException)

```

Comment:
Can you provide a little more info on your repro steps? It doesn't seem to happen on my machine when I call `gvfs clone`, or on any of our build agents, since they can all clone successfully. Any idea what would be causing this exception to get thrown in your environment?

That being said, I think this change is safe, since if we're not stopping, all we do is rethrow the exception, so for the most part the behavior should be unchanged from before.

Reply:
I'm yet to test this on Windows, but to me it happens on Mac all the time, when I do GVFS clone. The named pipe server, created in `CloneVerb.TryClone`, has already been disposed, for some reason `NamedPipeServer.OnNewConnection` is invoked (didn't dig deep enough to understand why) and once `pipe.EndWaitForConnection(ar);` is called - SocketException is thrown. My guess is that it might be a bug/difference in the underlying dotnet implementation, where on Windows it appropriately throws ObjectDisposedException and on Mac it ends up trying to access a closed socket. As far as I can tell, I'm not doing anything special.

Reply:
Just confirmed, that this is not happening with a windows client (same server).